src/shared/install.c

```diff
@@ -1439,6 +1440,18 @@ static int install_info_traverse(
                                 if (r < 0)
                                         return r;
 
+                                if (streq(buffer, i->name)) {
+
+                                        /* We filled in the instance, and the target stayed the same? If so, then let's
+                                         * honour the link as it is. */
+
+                                        r = install_info_follow(c, i, paths->root_dir, flags, true);
+                                        if (r < 0)
+                                                return r;
```

Comment:
It is normally an error to have a symlink to itself, but not in the systemd enable-by-symlink syntax. Indeed, we allow symlinks that point to nothing, we might just as well allow symlinks which loop.